scenes/ink_combat/ink_drinker/ink_drinker.gd

```diff
@@ -0,0 +1,65 @@
+class_name InkDrinker
+extends CharacterBody2D
+
+const INK_BLOB: PackedScene = preload("res://scenes/ink_combat/ink_blob/ink_blob.tscn")
+const BIG_SPLASH: PackedScene = preload("res://scenes/ink_combat/big_splash/big_splash.tscn")
+
+@export var odd_shoot: bool = false
+
+var health: float = 1.
+@onready var timer: Timer = %Timer
+
+@onready var ink_blob_marker: Marker2D = %InkBlobMarker
+@onready var hit_box: Area2D = %HitBox
+@onready var animated_sprite_2d: AnimatedSprite2D = %AnimatedSprite2D
+@onready var animation_player: AnimationPlayer = %AnimationPlayer
+@onready var health_bar: ProgressBar = %HealthBar
+
+
+func _ready() -> void:
+	timer.timeout.connect(_on_timeout)
+	hit_box.body_entered.connect(_on_got_hit)
+	if odd_shoot:
+		await get_tree().create_timer(timer.wait_time / 2).timeout
+	timer.start()
+
+
+func _on_timeout() -> void:
+	var player: FightingPlayer = get_tree().get_first_node_in_group("player")
+	if not is_instance_valid(player):
+		# or player.is_queued_for_deletion():
+		return
+	animated_sprite_2d.play(&"attack anticipation")
+	await animated_sprite_2d.animation_looped
+	animated_sprite_2d.play(&"attack")
+	var ink_blob: InkBlob = INK_BLOB.instantiate()
+	ink_blob.direction = player.global_position - ink_blob_marker.global_position
+	ink_blob.ink_color_name = randi_range(0, 3)
+	ink_blob.global_position = (
+		ink_blob_marker.global_position + ink_blob.direction.normalized() * 20.
+	)
+	Engine.get_main_loop().current_scene.add_child(ink_blob)
+	await animated_sprite_2d.animation_looped
+	animated_sprite_2d.play(&"idle")
+
+
+func _on_got_hit(body: Node2D) -> void:
+	if body is InkBlob and not body.can_hit_enemy:
+		return
+	body.queue_free()
+	animation_player.play(&"got hit")
+	health -= 0.05
+	health_bar.value = clamp(health, 0., 1.)
```

Comment:
It might be just a style preference, but I usually use move_toward for these cases. It makes sure the final value doesn't go past the destination (0.)
```suggestion
	health = move_toward(health, 0., 0.05)
```